doc/feature/ReqRep.md

```diff
@@ -0,0 +1,112 @@
+Request-Reply {#reqrep}
+============
+
+Certain operations are by necessity synchronous, RPC-like exchanges. This
+specification describes this client-server API based on zmq req-rep (dealer/router) sockets.
+
+## Requirements
+
+* Asynchronous remote procedure call from a Client to a Server
+
+## API
+
+    using ReplyFunc = std::function<void(const void*, size_t)>;
```

Comment:
Refresh my memory, do Zerobuf objects check the type ID during deserialization? Otherwise we are lacking type safety here, however I don't have a good suggestion to fix it (except add the type check in Zerobuf).
In any case, if the server doesn't use Zerobuf or some other typesafe serialization engine, we are in trouble.